src/Http/Http.Abstractions/src/HttpProtocol.cs

```diff
@@ -0,0 +1,87 @@
+        // and allow us to optimize comparisons when these constants are used.
+
+        // Please do NOT change these to 'const'
+        public static readonly string Http10 = "HTTP/1.0";
+        public static readonly string Http11 = "HTTP/1.1";
+        public static readonly string Http2 = "HTTP/2";
+        public static readonly string Http3 = "HTTP/3";
+
+        /// <summary>
+        /// Returns a value that indicates if the HTTP request protocol is HTTP/1.0. 
+        /// </summary>
+        /// <param name="protocol">The HTTP request protocol.</param>
+        /// <returns>
+        /// <see langword="true" /> if the protocol is HTTP/1.0; otherwise, <see langword="false" />.
+        /// </returns>
+        public static bool IsHttp10(string protocol)
+        {
+            return object.ReferenceEquals(Http10, protocol) || StringComparer.OrdinalIgnoreCase.Equals(Http10, protocol);
+        }
+
+        /// <summary>
+        /// Returns a value that indicates if the HTTP request protocol is HTTP/1.1. 
+        /// </summary>
+        /// <param name="protocol">The HTTP request protocol.</param>
+        /// <returns>
+        /// <see langword="true" /> if the protocol is HTTP/1.1; otherwise, <see langword="false" />.
+        /// </returns>
+        public static bool IsHttp11(string protocol)
+        {
+            return object.ReferenceEquals(Http11, protocol) || StringComparer.OrdinalIgnoreCase.Equals(Http11, protocol);
+        }
+
+        /// <summary>
+        /// Returns a value that indicates if the HTTP request protocol is HTTP/2. 
+        /// </summary>
+        /// <param name="protocol">The HTTP request protocol.</param>
+        /// <returns>
+        /// <see langword="true" /> if the protocol is HTTP/2; otherwise, <see langword="false" />.
+        /// </returns>
+        public static bool IsHttp2(string protocol)
+        {
+            return object.ReferenceEquals(Http2, protocol) || StringComparer.OrdinalIgnoreCase.Equals(Http2, protocol);
+        }
+
+        /// <summary>
+        /// Returns a value that indicates if the HTTP request protocol is HTTP/3. 
+        /// </summary>
+        /// <param name="protocol">The HTTP request protocol.</param>
+        /// <returns>
+        /// <see langword="true" /> if the protocol is HTTP/3; otherwise, <see langword="false" />.
+        /// </returns>
+        public static bool IsHttp3(string protocol)
+        {
+            return object.ReferenceEquals(Http3, protocol) || StringComparer.OrdinalIgnoreCase.Equals(Http3, protocol);
+        }
+
+        /// <summary>
+        /// Gets the HTTP request protocol for the specified <see cref="Version"/>.
+        /// </summary>
+        /// <param name="version">The version.</param>
+        /// <returns>A HTTP request protocol.</returns>
+        public static string GetProtocol(this Version version) => version switch
+        {
+            { Major: 2, Minor: 0 } => Http2,
```

Comment:
And HTTP/3?

Reply:
Can this inline switch use Version instances like System.Net.HttpVersion.Version20?

Reply:
I tried. Patterns need to be a const